extensions/ql-vscode/test/vscode-tests/activated-extension/databases/db-panel.test.ts

```diff
@@ -1,34 +1,55 @@
-import { window } from "vscode";
+import { authentication, window } from "vscode";
 
-import { readJson } from "fs-extra";
+import { readJson, writeJson } from "fs-extra";
 import * as path from "path";
 import {
   DbConfig,
   SelectedDbItemKind,
 } from "../../../../src/databases/config/db-config";
 import {
   AddListQuickPickItem,
+  CodeSearchQuickPickItem,
   RemoteDatabaseQuickPickItem,
 } from "../../../../src/databases/ui/db-panel";
 import { DbListKind } from "../../../../src/databases/db-item";
-import { createDbTreeViewItemSystemDefinedList } from "../../../../src/databases/ui/db-tree-view-item";
-import { createRemoteSystemDefinedListDbItem } from "../../../factories/db-item-factories";
+import {
+  createDbTreeViewItemSystemDefinedList,
+  createDbTreeViewItemUserDefinedList,
+} from "../../../../src/databases/ui/db-tree-view-item";
+import {
+  createRemoteSystemDefinedListDbItem,
+  createRemoteUserDefinedListDbItem,
+} from "../../../factories/db-item-factories";
 import { DbConfigStore } from "../../../../src/databases/config/db-config-store";
 import { getActivatedExtension } from "../../global.helper";
 import { createVSCodeCommandManager } from "../../../../src/common/vscode/commands";
 import { AllCommands } from "../../../../src/common/commands";
+import { MockGitHubApiServer } from "../../../../src/variant-analysis/gh-api/mocks/mock-gh-api-server";
+import { sleep } from "../../../../src/common/time";
+import { createDbConfig } from "../../../factories/db-config-factories";
 
 jest.setTimeout(60_000);
 
 describe("Db panel UI commands", () => {
+  // This test has some serious problems:
```

Comment:
Thanks for writing a comment to explain things, I should have done that when writing the tests because even I forgot about these limitations.

I just want to call out that this was intentional and the only way I could think of to add some integration tests for the commands without changing the actual implementation (not the test code) somehow. However, since it's turning out hard to add more tests, perhaps we do need to make some changes to the implementation to help us out a bit (not necessarily part of your work).